service/discovery/openstack/openstack.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2024 Fraunhofer AISEC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//           $$\                           $$\ $$\   $$\
+//           $$ |                          $$ |\__|  $$ |
+//  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+// $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+// $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+// $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+// \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+//  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+//
+// This file is part of Clouditor Community Edition.
+
+package openstack
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"clouditor.io/clouditor/v2/api/discovery"
+	"clouditor.io/clouditor/v2/api/ontology"
+	"clouditor.io/clouditor/v2/internal/config"
+	"clouditor.io/clouditor/v2/internal/util"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/openstack"
+	"github.com/gophercloud/gophercloud/v2/pagination"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	StorageComponent = "storage"
+	ComputeComponent = "compute"
+	NetworkComponent = "network"
+	RegionName       = "OS_REGION_NAME"
+)
+
+var (
+	log *logrus.Entry
+
+	ErrConversionProtobufToAuthOptions = errors.New("could not convert protobuf value to openstack.authOptions")
+	ErrGettingAuthOptionsFromEnv       = errors.New("error getting auth options from environment")
+	ErrCouldNotAuthenticate            = errors.New("could not authenticate to Azure")
+	ErrGettingNextPage                 = errors.New("error getting next page")
+	ErrNoCredentialsConfigured         = errors.New("no credentials were configured")
+)
+
+type openstackDiscovery struct {
+	csID     string
```

Comment:
I guess cs comes from cloud service which is now "certification target". We could think about changing this abbreviation (in another issue/PR)